File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {

    const dictionary: {[key: number]:number} = {};

    for (let i = 0; i < nums.length; i++){
        dictionary[nums[i]] = i;
    }

    for (let j = 0; j < nums.length; j++){
        const comp = target - nums[j];

        if (comp in dictionary && dictionary[comp] !== j){
            return [j, dictionary[comp]]
        }
    }
    return null
    
    };

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        diction = {}

        for i in range(len(nums)):
            diction[nums[i]] = i

        for i in range(len(nums)):

            subs = target - nums[i]
            if subs in diction and i != diction[subs]:
                return i,diction[subs]
